apps/generator/lib/generator.js

```diff
@@ -130,17 +130,32 @@ class Generator {
     /** @type {Object} The template parameters. The structure for this object is based on each individual template. */
     this.templateParams = {};
     Object.keys(templateParams).forEach(key => {
-      const self = this;
       Object.defineProperty(this.templateParams, key, {
         enumerable: true,
-        get() {
-          if (!self.templateConfig.parameters || !self.templateConfig.parameters[key]) {
-            throw new Error(`Template parameter "${key}" has not been defined in the package.json file under generator property. Please make sure it's listed there before you use it in your template.`);
+        get: () => {
+          // Using ?. to check if parameters and the key exist
+          // This is shorter and does the same thing as the old if statements
+          if (this.templateConfig.parameters?.[key] == null) {
```

Comment:
Here there's one condition also exists `!self.templateConfig.parameters` pls. add that also in validation.

Reply:
Please check PR description where I mention the issue that I solved in this PR sonarcloud highlight the issue that Prefer using an optional chain expression instead, as it's more concise and easier to read. that's what I did if i implement one more codition `!self.templateConfig.parameters` then this PR makes no sense here I hope you understand 

This change makes the code more concise and easier to read while maintaining the same functionality. It checks if either the parameters object doesn't exist or if the specific key is not defined within it.